app/helpers/application_helper.rb

```diff
@@ -475,11 +475,20 @@ def accessible_resource_records(resource, order = :name)
   end
 
   def accessible_resource(obj, resource, order = :name, association: resource)
-    list = accessible_resource_records(resource, order).to_a
```

Comment:
Can we cache accessible records per-request? We are reusing the same list many times inside a form.
Good example would be NICs, where for each rendered NIC, we render list of domains that shouldn't change between NICs in the same request.

I am thinking about `class => resources_list` kind of cache here, so we won't need to implement it for each field in `nics/_base_form.html.erb`.

Reply:
@ShimShtein  we do have a plan to use a react implementation for select2, in fact the switcher (from breadcrumbs switcher) is almost a  select with search ability already.

@glekner have you started to work on select2 component ?